### DISCUSSION TITLE
Flow Protocol: Sending a Request Only If the Prior Request Matcher Fails

### DISCUSSION THREAD

**Discussion by ayushkr12** [projectdiscovery/nuclei#5613] [Q&A]
From the [Flow Protocol](https://docs.projectdiscovery.io/templates/protocols/flow) documentary i know i can control the flow of 2 requests in such a way if i want to run the `http(2)` request only if `http(1)` request `matcher` returned `true`. But how do i control the flow to do the opposite i.e. if `http(1)` matcher was `true` do not run `http(2)`. Below is my code

```yaml
id: wp-detect

info:
  name: WordPress Detect
  author: dat-ayush
  severity: info
  description: Detects if a domain is running WordPress
  tags: cms,wordpress

flow: http(1) && http(2)

http:
  - method: GET
    path:
      - "{{RootURL}}"

    redirects: true
    max-redirects: 3

    matchers:
      - type: regex
        regex:
          - "(?i)wordpress"

  - method: GET
    path:
      - "{{RootURL}}/wp-admin"
    
    redirects: true
    max-redirects: 3

    matchers:
      - type: word
        words:
          - 'Powered by WordPress'
```

endpoints

- `/` 
-  `/wp-admin` 
 
so basically i only want to request `wp-admin` if `/` matcher fails i.e. when `wordpress` is not found in it's body. I looked everywhere in the flow protcol documentary but was unable to find this feature, I would be grateful if anyone could help me with it. Many Thanks!

**Comment by ayushkr12** [comment]
copy pasted the same question on chatGPT and it's solution worked!: `flow: http(1) || http(2)`
